### PR TITLE
JSDoc in Get Started Section

### PR DESCRIPTION
**Pull request by Sarfraz-droid**
#### Checklist

Issue : #3725 

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
Please provide a short description for the `JSDoc` param and function.
e.g. what is it? document reference?

**Comment by Sarfraz-droid** [comment]
> Please provide a short description for the `JSDoc` param and function. e.g. what is it? document reference?

Should I provide? on top file or everywhere it is used?

**Comment by climba03003** [comment]
> > Please provide a short description for the `JSDoc` param and function. e.g. what is it? document reference?
> 
> Where should I provide? on top file or everywhere it is used?

For example
```
/**
 * a plugin that provide encapsulated routes
 * @param {FastifyInstance} fastify encapsulated fastify instance
 * @param {Object} options plugin options, may refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
 * */
async function routes (fastify, options) {
```

**Comment by Fdawgs** [comment]
This does not resolve #3725. @RafaelGSS was talking about `@type` jsdoc tags: https://github.com/fastify/fastify/issues/3725#issuecomment-1049922940

**Comment by Sarfraz-droid** [comment]
@Fdawgs What typename should i attach to ?

```
const fastify = Fastify({
  logger: true
})
```

**Comment by Sarfraz-droid** [comment]
@climba03003 can you have a look the changes? I have added short descriptions for JSDoc

File: docs/Guides/Getting-Started.md [hunk 1]
```diff
@@ -403,6 +440,10 @@ Schema](https://json-schema.org/).
 
 Let's look at an example demonstrating validation for routes:
 ```js
+/** 
+ * @type {Object}
```
Comment (climba03003) [inline docs/Guides/Getting-Started.md:444]:
```suggestion
 * @type {import('fastify').RouteShorthandOptions}
```

File: docs/Guides/Getting-Started.md [hunk 2]
```diff
@@ -435,6 +476,10 @@ JSON bodies and to serialize JSON output.
 To speed up JSON serialization (yes, it is slow!) use the `response` key of the
 schema option as shown in the following example:
 ```js
+/** 
+ * @type {Object}
```
Comment (climba03003) [inline docs/Guides/Getting-Started.md:480]:
```suggestion
 * @type {import('fastify').RouteShorthandOptions}
```

File: docs/Guides/Getting-Started.md [hunk 3]
```diff
@@ -468,6 +513,9 @@ request](../Reference/Request.md) object at `request.body`.
 The following example returns the parsed body of a request back to the client:
 
 ```js
+/**
+ * @type {Object}
```
Comment (climba03003) [inline docs/Guides/Getting-Started.md:517]:
```suggestion
 * @type {import('fastify').RouteShorthandOptions}
```

File: docs/Guides/Getting-Started.md [hunk 4]
```diff
@@ -150,6 +156,10 @@ fastify.listen({ port: 3000 }, function (err, address) {
 
 ```js
 // CommonJs
+/**
+ *
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:160]:
```suggestion
```

File: docs/Guides/Getting-Started.md [hunk 5]
```diff
@@ -150,6 +156,10 @@ fastify.listen({ port: 3000 }, function (err, address) {
 
 ```js
 // CommonJs
+/**
+ * 
+ * @type {import('fastify').FastifyInstance} Instance of Fastify
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:162]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 6]
```diff
@@ -168,6 +178,12 @@ fastify.listen({ port: 3000 }, function (err, address) {
 ```js
 // our-first-route.js
 
+/**
+ * Encapsulates the routes
+ * @param {FastifyInstance} fastify  Encapsulated Fastify Instance
+ * @param {Object} options plugin options, may refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
+
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:186]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 7]
```diff
@@ -74,6 +74,9 @@ fastify.get('/', async (request, reply) => {
   return { hello: 'world' }
 })
 
+/**
+ * Run the server!
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:79]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 8]
```diff
@@ -133,6 +136,9 @@ declaration](../Reference/Routes.md) docs).
 // ESM
 import Fastify from 'fastify'
 import firstRoute from './our-first-route'
+/**
+ * @type {import('fastify').FastifyInstance} Instance of Fastify
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:141]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 9]
```diff
@@ -208,6 +224,9 @@ import Fastify from 'fastify'
 import dbConnector from './our-db-connector'
 import firstRoute from './our-first-route'
 
+/**
+ * @type {import('fastify').FastifyInstance} Instance of Fastify
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:229]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 10]
```diff
@@ -225,6 +244,9 @@ fastify.listen({ port: 3000 }, function (err, address) {
 
 ```js
 // CommonJs
+/**
+ * @type {import('fastify').FastifyInstance} Instance of Fastify
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:249]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 11]
```diff
@@ -248,6 +270,11 @@ fastify.listen({ port: 3000 }, function (err, address) {
 import fastifyPlugin from 'fastify-plugin'
 import fastifyMongo from 'fastify-mongodb'
 
+/**
+ * 
+ * @param {FastifyInstance} fastify
+ * @param {Object} options
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:277]:
```suggestion
/**
 * @param {FastifyInstance} fastify
 * @param {Object} options
 */
```

File: docs/Guides/Getting-Started.md [hunk 12]
```diff
@@ -468,6 +513,9 @@ request](../Reference/Request.md) object at `request.body`.
 The following example returns the parsed body of a request back to the client:
 
 ```js
+/**
+ * @type {Object}
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:518]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 13]
```diff
@@ -435,6 +476,10 @@ JSON bodies and to serialize JSON output.
 To speed up JSON serialization (yes, it is slow!) use the `response` key of the
 schema option as shown in the following example:
 ```js
+/** 
+ * @type {Object} 
+ * @const
+ *  */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:482]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 14]
```diff
@@ -403,6 +440,10 @@ Schema](https://json-schema.org/).
 
 Let's look at an example demonstrating validation for routes:
 ```js
+/** 
+ * @type {Object} 
+ * @const
+ *  */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:446]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md [hunk 15]
```diff
@@ -294,12 +336,7 @@ async function routes (fastify, options) {
   })
 
   fastify.get('/animals/:animal', async (request, reply) => {
-    const result = await collection.findOne({ animal: request.params.animal })
-    if (!result) {
-      throw new Error('Invalid value')
-    }
-    return result
-  })
+    const rer
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:339]: Where'd this go?

File: docs/Guides/Getting-Started.md [hunk 16]
```diff
@@ -278,6 +314,12 @@ module.exports = fastifyPlugin(dbConnector)
 
 **our-first-route.js**
 ```js
+/**
+ *  a plugin that provide encapsulated routes 
+ * @param {FastifyInstance} fastify encapsulated fastify instance
+ * @param {Object} options plugin options, may refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
+
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:322]:
```suggestion
/**
 * A plugin that provide encapsulated routes 
 * @param {FastifyInstance} fastify encapsulated fastify instance
 * @param {Object} options plugin options, refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
 */
```

File: docs/Guides/Getting-Started.md [hunk 17]
```diff
@@ -262,8 +289,17 @@ module.exports = fastifyPlugin(dbConnector)
...
+ * @type {import('fastify-plugin').FastifyPlugin}
+ * */
 const fastifyPlugin = require('fastify-plugin')
 
+
+/**
+ * Connects to a MongoDB database
+ * @param {FastifyInstance} fastify Encapsulated Fastify Instance
+ * @param {Object} options plugin options, may refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:302]:
```suggestion
/**
 * Connects to a MongoDB database
 * @param {FastifyInstance} fastify Encapsulated Fastify Instance
 * @param {Object} options plugin options, refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
 */
```

File: docs/Guides/Getting-Started.md [hunk 18]
```diff
@@ -262,8 +289,17 @@ module.exports = fastifyPlugin(dbConnector)
 
 ```js
 // CommonJs
+/**
+ * @type {import('fastify-plugin').FastifyPlugin}
+ * */
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:294]:
```suggestion
 */
```

File: docs/Guides/Getting-Started.md (hunk 15 above)
Comment (Sarfraz-droid) [reply docs/Guides/Getting-Started.md]: Sorry I might have removed it by mistake.. I have reverted these changes

File: docs/Guides/Getting-Started.md [hunk 19]
```diff
@@ -168,6 +177,11 @@ fastify.listen({ port: 3000 }, function (err, address) {
 ```js
 // our-first-route.js
 
+/**
+ * Encapsulates the routes
+ * @param {FastifyInstance} fastify  Encapsulated Fastify Instance
+ * @param {Object} options plugin options, may refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
```
Comment (Fdawgs) [inline docs/Guides/Getting-Started.md:183]:
```suggestion
 * @param {Object} options plugin options, refer to https://www.fastify.io/docs/latest/Reference/Plugins/#plugin-options
```

**Review by 3imed-jaberi — APPROVED** [review]
+1 ✅

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Fdawgs** [comment]
@climba03003 mind re-reviewing when you get a sec, please?

**Review by climba03003 — APPROVED** [review]

**Pull request merged**